### DISCUSSION TITLE
Source code searching

### DISCUSSION THREAD

**Discussion by sbcguard** [codefori/vscode-ibmi#2087] [Ideas]
Not really an issue but a question/idea. Currently in RDi, we use iSphere to search source code. Is there something similar to that in Code for i, or plans for something like that. I work in a lot of older COBOL programs and the like, and the source code search is often valuable in finding where a program is called from and various other quick search needs.
Thanks,
AJ

<hr />

⚠️ **REMOVE THIS LINE AND ANY SENSITIVE INFORMATION BELOW!** ⚠️

|Context|Version|
|-|-|
|Code for IBM i version|2.10.4|
|Visual Studio Code version|1.89.1|
|Operating System|win32_x64|

<details>
<summary>Active extensions</summary>

```
COBOL (cobol): 10.5.29
Code Runner (code-runner): 0.12.2
Code Spell Checker (code-spell-checker): 3.0.1
Code for IBM i Walkthroughs (vscode-ibmi-walkthroughs): 0.5.0
CodeTour (codetour): 0.0.59
Db2 for IBM i (vscode-db2i): 1.0.0
ES7+ React/Redux/React-Native snippets (es7-react-js-snippets): 4.4.3
ESLint (vscode-eslint): 2.4.4
EditorConfig for VS Code (EditorConfig): 0.16.4
Emmet (emmet): 1.0.0
Error Lens (errorlens): 3.18.0
Git (git): 1.0.0
Git Base (git-base): 1.0.0
Git Project Manager (git-project-manager): 1.8.2
GitHub (github): 0.0.1
GitHub Authentication (github-authentication): 0.0.2
GitLens — Git supercharged (gitlens): 15.0.4
IBM i Debug (ibmidebug): 1.0.0
JSON Language Features (json-language-features): 1.0.0
Live Server (LiveServer): 5.7.9
Live Share (vsliveshare): 1.0.5918
Material Icon Theme (material-icon-theme): 5.3.0
Merge Conflict (merge-conflict): 1.0.0
Microsoft Account (microsoft-authentication): 0.0.1
Node Debug Auto-attach (debug-auto-launch): 1.0.0
Prettier - Code formatter (prettier-vscode): 10.4.0
TODO Highlight (vscode-todo-highlight): 1.0.5
Tabnine AI (tabnine-vscode): 3.104.0
TypeScript and JavaScript Language Features (typescript-language-features): 1.0.0
```
</details>

<hr />

<details>
<summary>Remote system</summary>

|Setting|Value|
|-|-|
|IBM i OS|n/a|
|Tech Refresh|n/a|
|CCSID Origin|65535|
|Runtime CCSID|65535|
|Default CCSID|37|
|SQL|Enabled
|Source dates|Disabled

### Enabled features

|/QOpenSys/pkgs/bin|/usr/bin|/QSYS.lib/ILEDITOR.lib||/QSYS.LIB|/QIBM/ProdData/IBMiDebugService/bin|
|-|-|-|-|-|-|
|bash|attr|GETNEWLIBL.PGM|ILEDITOR.GETMBRINFO|QZDFMDB2.PGM|startDebugService.sh|
|git|iconv|||||
|grep|setccsid|||||
|ls||||||
|md5sum||||||
|sort||||||
|stat||||||
|tar||||||
</details>

<details>
<summary>Shell env</summary>

```bash
```
</details>

<details>
<summary>Variants</summary>

```json
{
  "american": "#@$",
  "local": "#@$"
}
```
</details>

<details>
<summary>Errors</summary>

```json
[
  {
    "command": "/QOpenSys/usr/bin/qsh",
    "code": 1,
    "stderr": "CPF2111:  Library ILEDITOR already exists.",
    "cwd": "/home/ITAJ"
  },
  {
    "command": "export BUILDLIB=\"BC001\" && export CURLIB=\"BC001\" && export USERNAME=\"ITAJ\" && export HOST=\"BOONEDEV\" && export HOME=\"/home/ITAJ\" && export WORKDIR=\"/home/ITAJ\" && export LIBLS=\"TAATOOL QGPL QGDDM QTEMP RVILIB RVICUST MMAIL BC001 ITA05\" && env",
    "code": 1,
    "stderr": "bsh: BUILDLIB=BC001: is not an identifier",
    "cwd": "/home/ITAJ"
  },
  {
    "command": "export BUILDLIB=\"BC001\" && export CURLIB=\"BC001\" && export USERNAME=\"ITAJ\" && export HOST=\"BOONEDEV\" && export HOME=\"/home/ITAJ\" && export WORKDIR=\"/home/ITAJ\" && export LIBLS=\"TAATOOL QGPL QGDDM QTEMP RVILIB RVICUST MMAIL BC001 ITA05\" && env",
    "code": 1,
    "stderr": "bsh: BUILDLIB=BC001: is not an identifier",
    "cwd": "/home/ITAJ"
  }
]
```
</details>

**Comment by sebjulliand** [comment]
Could this be what  you're looking for?
![image](https://github.com/codefori/vscode-ibmi/assets/11096890/31c87e52-36fb-4310-a685-08007ae5202b)

**Comment by sbcguard** [comment]
Eh, not exactly, with the RDi iSphere source file search you can specify the LIB you want to search in, or search all libs. That will only look for references within that source member.

**Comment by sbcguard** [comment]
For example,

![image](https://github.com/codefori/vscode-ibmi/assets/123800680/45c030c9-545d-4d57-beb9-6a173ad82221)
![image](https://github.com/codefori/vscode-ibmi/assets/123800680/ce3f9d64-ff9f-48d8-82f9-78f00653c476)

This will search all libraries and all source members for a value

**Reply by acejoh** [reply]
+1 for this feature!
